service/service.go

```diff
@@ -107,21 +112,54 @@ func (s *Service) CreateUser(ctx context.Context, req *pb.UserRequest) (*pb.User
 	user.Password = hashedPassword
 	user.IsVerified = false
 
-	// insert into DB
+	// create unique email token
+	token, err := generateEmailToken()
+	if err != nil {
+		logger.Error("CreateUser generateEmailToken:", err.Error())
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+
+	// insert user into DB
 	command := `
-				INSERT INTO user_account(
+				INSERT INTO user_svc.accounts(
 					uuid, first_name, last_name, email, password, organization, created_date, is_verified
 				) VALUES($1, $2, $3, $4, $5, $6, $7, $8)
 				`
 	_, err = postgresDB.Exec(command, user.GetUuid(), user.GetFirstName(), user.GetLastName(),
 		user.GetEmail(), user.GetPassword(), user.GetOrganization(), time.Now().UTC(), user.GetIsVerified())
 
 	if err != nil {
-		logger.Error("CreateUser Exec INSERT:", err.Error())
+		logger.Error("CreateUser INSERT user_svc.accounts:", err.Error())
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	logger.Info("Success INSERT new user:", user.GetFirstName(), user.GetLastName(), user.GetUuid())
+
+	// insert token into db
+	command = `INSERT INTO user_svc.pending_tokens(token, created_date, uuid) VALUES($1, $2, $3)`
+	_, err = postgresDB.Exec(command, token, time.Now().UTC(), user.GetUuid())
+	if err != nil {
+		logger.Error("CreateUser INSERT user_svc.pending_tokens:", err.Error())
```

Comment:
starting from this block on and onwards (refer to the images I posted), its a huge chunk I can't test, so I am wondering if I should move this to a utility or db function so I can test that instead.

Reply:
I figure that we can add a task to make utility functions that generate query strings.

Reply:
Add a `// TO-DO https://github.com/hwsc-org/hwsc-user-svc/issues/21`